openlabcmd/openlabcmd/plugins/nodepool/orphanresource.py

```diff
@@ -0,0 +1,120 @@
+import copy
+import subprocess
+
+from openlabcmd.plugins.base import Plugin
+
+SERVER_WHITE_LIST = [
+    '%s-openlab-zuul', '%s-openlab-nodepool', '%s-openlab-zookeeper']
+
+
+class OrphanResourcePlugin(Plugin):
+
+    # This Plugin is only for checking, as it's a dangerous action for each.
+    ptype = 'nodepool'
+    name = 'OrphanResource'
+
+    def check(self):
+        self.failed = False
+        if self.reasons:
+            self.reasons.pop()
+        server_white_list = [elemnt % self.cloud
+                             for elemnt in copy.deepcopy(SERVER_WHITE_LIST)]
+
+        # For floatingip, we will check whether a fip didn't associated with a
+        # neutron port.
+        get_fip_cmd = ("openstack --os-cloud %s floating ip list "
+                       "-c 'ID' -c 'Floating IP Address' -c 'Port' | "
+                       "egrep '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-"
+                       "[a-z0-9]{4}-[a-z0-9]{12}' | awk '{print $2, $4, $6}' "
+                       "| grep None | awk '{print $1}'")
+        # For volume, we just check the available status volumes, as we cannot
+        # know whether a in-use volume is used for any purpose.
+        get_volume_cmd = (
+            "openstack --os-cloud %s volume list -c 'ID' -c 'Status' | "
+            "egrep '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-"
+            "[a-z0-9]{12}' | grep available | awk '{print $2}'")
+        # For servers, we will get all servers for a provider, and skip the
+        # online openlab env servers.
+        get_server_cmd = (
+            "openstack --os-cloud %s server list -c 'ID' -c 'Name' | "
+            "egrep '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-"
+            "[a-z0-9]{12}' | awk '{print $2, $4}'")
+        # Get the full map servers from nodepool, we will detemine whether a
+        # server is orphan based on this data.
+        get_nodepool_map_cmd = (
+            "nodepool list | grep openlab | awk '{print $2, $4, $8, $10}' "
+            "| sort -t ' ' -k2")
+        ret, res = subprocess.getstatusoutput(get_nodepool_map_cmd)
+
+        mapping = {}
+        for line in res.split('\n'):
+            elements = line.split(' ')
+            if elements[1] not in mapping:
+                mapping[elements[1]] = {
+                    'nodepool_compute_mapping':
+                        [{'nodepool_id': elements[0],
```

Comment:
This seems never used? and the name is somehow missleading, could we just don't set this?

Reply:
correct, I will remove this.